tests/worker/test_workflow.py

```diff
@@ -789,6 +899,255 @@ async def test_workflow_cancel_multi(client: Client):
         ]
 
 
+@workflow.defn
+class CancelUnsentWorkflow:
+    @workflow.run
+    async def run(self) -> None:
+        # Timer
+        def raise_error():
+            raise RuntimeError("should not get here")
+
+        timer_handle = asyncio.get_running_loop().call_later(1, raise_error)
+        timer_handle.cancel()
+
+        async def wait_timer():
+            await timer_handle
+
+        await self.wait_and_swallow(wait_timer())
+
+        # Start activity
+        activity_handle = workflow.start_activity(
+            wait_cancel, schedule_to_close_timeout=timedelta(seconds=5)
+        )
+        activity_handle.cancel()
+        await self.wait_and_swallow(activity_handle)
+
+        # Execute activity
+        activity_task = asyncio.create_task(
+            workflow.execute_activity(
+                wait_cancel, schedule_to_close_timeout=timedelta(seconds=5)
+            )
+        )
+        activity_task.cancel()
+        await self.wait_and_swallow(activity_task)
+
+        # Start local activity
+        activity_handle = workflow.start_local_activity(
+            wait_cancel, schedule_to_close_timeout=timedelta(seconds=5)
+        )
+        activity_handle.cancel()
+        await self.wait_and_swallow(activity_handle)
+
+        # Execute local activity
+        activity_task = asyncio.create_task(
+            workflow.execute_local_activity(
+                wait_cancel, schedule_to_close_timeout=timedelta(seconds=5)
+            )
+        )
+        activity_task.cancel()
+        await self.wait_and_swallow(activity_task)
+
+        # Start child
+        child_task1 = asyncio.create_task(
+            workflow.start_child_workflow(
+                LongSleepWorkflow.run, id=f"{workflow.info().workflow_id}_child1"
+            )
+        )
+        child_task1.cancel()
+        await self.wait_and_swallow(child_task1)
+
+        # Execute child
+        child_task2 = asyncio.create_task(
+            workflow.execute_child_workflow(
+                LongSleepWorkflow.run, id=f"{workflow.info().workflow_id}_child2"
+            )
+        )
+        child_task2.cancel()
+        await self.wait_and_swallow(child_task2)
+
+        # Sleep for a short bit to force another task to run so we know that
+        # workflow completion isn't saving us here
+        await asyncio.sleep(0.01)
+
+    async def wait_and_swallow(self, aw: Awaitable) -> None:
+        try:
+            await aw
+        except:
```

Comment:
Same as above

Reply:
Fine by me if we swallow that. This is test code.

Reply:
Fixed